src/Controller/AdminController.php

```diff
@@ -4,8 +4,127 @@
 
 class AdminController extends AbstractController
 {
-    public function course(): string
+    public function course(string $success = null): string
     {
-        return $this->twig->render('Admin/course.html.twig');
+        $errors = [];
+        $formData = [];
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            $formData = array_map('trim', $_POST);
+            $errors = $this->validateFormulary($formData);
+            $formData['duration'] = $this->intValue($formData['duration']);
+            if ($formData['duration'] === 0) {
+                $formData['duration'] = '';
+            }
+            $formData['capacity'] = $this->intValue($formData['capacity']);
+            if ($formData['capacity'] === 0) {
+                $formData['capacity'] = '';
+            }
+            if (empty($errors)) {
+                header('location: /Admin/course/success');
+            }
+        }
+        if ($success !== null) {
+            $success = 'Cours enregistré avec succès!';
+        }
+        return $this->twig->render('Admin/course.html.twig', [
+            'errors' => $errors,
+            'formulary' => $formData,
+            'success' => $success,
+        ]);
+    }
+
+    /**
+     * Test fields if exist, if not empty and test their constraints
+     *
+     * @param array $formData
+     * @return array
+     */
+    private function validateFormulary(array $formData): array
+    {
+        define('INPUTS', [
```

Comment:
this constant's name is not really descriptive

Reply:
yes correction done